### ISSUE TITLE
Make a separate CLI tool that wraps exiftool and shows a colorized diff

### ISSUE THREAD

**Issue by Gitoffthelawn** [szTheory/exifcleaner#165]
In addition to drag and drop, it would be great to support command-line parameters specifying a list of files to process.

Thanks!

**Comment by szTheory** [comment]
For command-line usage you're better off just using the `exiftool` command line tool directly, which ExifCleaner is just a GUI wrapper around.  Just run `exiftool --help` and it will give you usage examples at the bottom.

**Comment by Gitoffthelawn** [comment]
That's what I was thinking, but I really like how you provide the before/after stats so you can see that things worked.  Combined with #167, ExifCleaner would have big benefits over exiftool (I'm imagining the ExifCleaner window still being shown after it processes files obtained from the command line).

**Comment by szTheory** [comment]
I see your point, it's actually a pretty good idea to have a command line tool that only removes the metadata and shows a nice diff afterwards. Although I think it would be better to make a separate CLI tool project for that rather than try to shoehorn it into ExifCleaner. Maybe a Perl CLI command that just directly calls the exiftool library and shows the colored diff.

**Comment by Gitoffthelawn** [comment]
Hmmm.... I'm actually not imagining it as being CLI.  I'm imagining it as exactly what ExifCleaner does except instead of specifying the files to process via drag-and-drop, allowing them to be specified via the command line.

This way, at least in Windows, it is trivial for the user to add a context menu item to their file manager to send one or more files/folders directly to ExifCleaner.  Then, just by right-clicking on one or more files/folders (within the file manager) and selecting the appropriate menuitem, they will be sent to ExifCleaner to be processed.  Except allowing files/folders to be specified via a parameter, nothing else would need to change in ExifCleaner to allow this powerful functionality.

**Comment by szTheory** [comment]
Oh really? I didn't know that you could add context menu items in Windows if a program accepted command line arguments. I know there are at least 3 other desktop integration features that require CLI parameters, for example:
https://github.com/szTheory/exifcleaner/issues/38
https://github.com/szTheory/exifcleaner/issues/39
https://github.com/szTheory/exifcleaner/issues/40

So I will definitely consider it!

**Comment by MohammedReyiad** [comment]
good day sir 
i have tried to do so 
i was able to add context menu name and launch the app with it But was not able to make the app open the sent file location 
[ExifCleaner.txt](https://github.com/szTheory/exifcleaner/files/10068419/ExifCleaner.txt)
this is the registry file 
hope it helps
yours 
mohammedreyiad

**Comment by MohammedReyiad** [comment]
just rename file ext from txt to reg

**Comment by szTheory** [comment]
Keeping this open, with an honest note on why it hasn't moved.

Two things are tangled in this thread: CLI parameters (accept a list of file paths as arguments) and OS right-click context-menu integration built on top of them. The first is small. The second is where all the questions live. #161 has now been closed into this issue, so there's one place for both.

What would move it is a clear *visual* proposal of the behavior — what the context-menu entry says, what the user sees after clicking it when no window is open, where output goes, how failures surface, and what happens with a mixed selection of supported and unsupported files. Mockups or even rough sketches beat prose here.

To be clear about the appetite: I like tightly integrated quality-of-life features that make the app more seamless without expanding what it does. I'm wary of shipping a version of this that behaves in ways people can't predict. A concrete, illustrated proposal would get a real look.

**Comment by webguyio** [comment]
I agree that for full command line support, people might as well just use https://exiftool.org/ directly. But maybe there's a little room for CLI support here too?

But as a mainly GUI wrapper, context menu support for ExifCleaner would be a nice-to-have.

It looks doable without code signing or native code by using each OS's built-in integration points:

**Windows**
- NSIS installer writes a shell verb to the registry (e.g. `HKCU\Software\Classes\*\shell\ExifCleaner`) and removes it on uninstall.
- Add `app.requestSingleInstanceLock()` with a `second-instance` handler so multi-file selections go to one window.
- Note: installer build only (not portable), and Windows 11 shows it under "Show more options."

**macOS**
- Add `fileAssociations` in electron-builder so ExifCleaner appears under **Open With**.
- Handle `app.on( 'open-file' )` early in the main process and queue paths until the window is ready.

**Linux**
- Add `MimeType` to the generated `.desktop` file for **Open With**.
- Optionally, `.deb`/`.rpm` post-install scripts can add Dolphin and Nemo actions.
- Note: AppImage can't register these.

On every platform the app-side change is the same: feed incoming file paths into the existing clean queue. Since cleaning overwrites files by default, it might be worth labeling the menu item clearly (for example "Clean Metadata") or gating auto-clean behind a setting.

PS: I realize migrating to something else other than Electron is highly unlikely and having to maintain extra features like this for 3 different operating systems for only minor efficiency improvements may simply not be worth it. As a fellow developer, your sanity is the most important thing of all to maintain!
